src/RoadRegistry.Editor.Projections/RoadSegmentDynamicAttributeSynchronizationBehavior.cs

```diff
@@ -0,0 +1,35 @@
+namespace RoadRegistry.Editor.Projections
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.EntityFrameworkCore;
+
+    public static class RoadSegmentDynamicAttributeSynchronizationBehavior
+    {
+        public static async Task Synchronize<T>(this DbSet<T> source,
+            Dictionary<int, T> currentSet,
+            Dictionary<int, T> nextSet,
+            Action<T, T> modifier) where T : class
+        {
+            var allKeys = new HashSet<int>(currentSet.Keys.Concat(nextSet.Keys));
+            foreach (var key in allKeys)
+            {
+                if (currentSet.TryGetValue(key, out var current) && nextSet.TryGetValue(key, out var next))
```

Comment:
I can further optimise this such that TryGet is only needed twice. I'll update tomorrow.